src/group_theory/submonoid/operations.lean

```diff
@@ -9,6 +9,7 @@ import group_theory.submonoid.basic
 import data.equiv.mul_add
 import algebra.group.prod
 import algebra.group.inj_surj
+import data.set.finite
```

Comment:
I'm a little wary of importing finiteness here. Is it worth having a `group_theory/finiteness` file for these new lemmas, and perhaps even a `submonoid.fg` definition?

Reply:
I don't know about the import, but I agree we need something about finitely generated group. In my case the groups are abelian, so I can use the API for submodules (in the additive case).

Reply:
This is now solved, using `monoid.fg`.